channeldb/invoices.go

```diff
@@ -546,6 +546,76 @@ func (d *DB) LookupInvoice(paymentHash [32]byte) (Invoice, error) {
 	return invoice, nil
 }
 
+// InvoiceWithPaymentHash is used to store an invoice and its corresponding
+// payment hash. This struct is only used to store results of
+// ChannelDB.FetchAllInvoicesWithPaymentHash() call.
+type InvoiceWithPaymentHash struct {
+	Invoice     Invoice
+	PaymentHash lntypes.Hash
+}
+
+// FetchAllInvoicesWithPaymentHash returns all invoices and their payment hashes
+// currently stored within the database. If the pendingOnly param is true, then
+// only unsettled invoices and their payment hashes will be returned, skipping
+// all invoices that are fully settled or canceled. Note that the returned
+// array is not ordered by add index.
+func (d *DB) FetchAllInvoicesWithPaymentHash(pendingOnly bool) (
+	[]InvoiceWithPaymentHash, error) {
+
+	var result []InvoiceWithPaymentHash
+
+	err := d.View(func(tx *bbolt.Tx) error {
+		invoices := tx.Bucket(invoiceBucket)
+		if invoices == nil {
+			return ErrNoInvoicesCreated
```

Comment:
Shouldn't it return an empty list in this case?

Reply:
Hmm this is weird, looks like I lost at least one commit during rebase?

Reply:
In my opinion it may be better to stay consistent with `FetchAllInvoices` and return nil and the error.

Reply:
`invoiceBucket` is created when a new database is created, but that isn't true for `invoiceIndexBucket`. `invoiceIndexBucket` isn't accessed in `FetchAllInvoices`, so the problem won't show there. 

I think the code as it currently is will generate an error with an empty database?

Reply:
Modified as discussed offline, and added a unit test as well.

Reply:
It now only returns an error if the`invoiceBucket` does not exists (as that indicates a real error)

Reply:
Returning nil happens in later commit, but should be done here.